### PR TITLE
[Bug-Fix] Do not copy UI's node_modules folder in ArangoShell (V8)

### PR DESCRIPTION
**Pull request by hkernbach**
### Scope & Purpose

Fix `node_modules` copy in arangosh. 
Quite close to the previous PR: https://github.com/arangodb/arangodb/pull/20478
But now "client" part. 

- [x] :hankey: Bugfix

### Checklist

- [x] :book: CHANGELOG entry made

### CODE REVIEW

File: client-tools/Shell/V8ShellFeature.cpp [hunk 1]
```diff
@@ -310,9 +310,20 @@ void V8ShellFeature::copyInstallationFiles() {
...
+      FileUtils::buildFilename("js", "apps", "system", "_admin", "aardvark",
+                               "APP", "react", "node_modules");
   std::regex const binRegex("[/\\\\]\\.bin[/\\\\]", std::regex::ECMAScript);
 
-  auto filter = [&nodeModulesPath, &nodeModulesPathVersioned,
+  auto filterPath = [](std::string_view normalizedPath,
+                       std::string_view filterPath) -> bool {
+    return (!filterPath.empty() && normalizedPath.size() >= filterPath.size() &&
+            normalizedPath.substr(normalizedPath.size() - filterPath.size(),
+                                  filterPath.size()) == filterPath);
```
Comment (jsteemann) [inline client-tools/Shell/V8ShellFeature.cpp:322]: Isn't this the same as `return !filterPath.empty() && normalizedPath.ends_with(filterPath);`?

**Comment by hkernbach** [comment]
Windows run can be found here:
- https://jenkins.arangodb.biz/job/arangodb-matrix-pr-windows/24123/

For some reason it did not get linked automatically here from jenkins.

**Review by jsteemann — CHANGES_REQUESTED** [review]
While this fix works, I would be in favor of not even trying to copy all these files when starting the arangosh.
What we actually only need to copy when starting arangosh is `js/client` and `js/common`. Maybe also `js/node`, but I don't know.
Neither `js/server` nor `js/apps` nor `js/actions` are needed to boot up the arangosh, so we should not even attempt to copy files in these directories.

**Comment by hkernbach** [comment]
Jenkins Windows Run: https://jenkins.arangodb.biz/job/arangodb-matrix-pr-windows/24124/

**Review by jsteemann — APPROVED** [review]
LGTM

**Comment by hkernbach** [comment]
Another Jenkins Windows Run: https://jenkins.arangodb.biz/job/arangodb-matrix-pr-windows/24136/

**Pull request merged**